renku/models/_jsonld.py

```diff
@@ -346,24 +351,25 @@ def from_jsonld(cls, data, __reference__=None, __source__=None):
 
         if __reference__:
             with with_reference(__reference__):
-                self = cls(**data_)
+                self = cls(client=client, **data_)
```

Comment:
I think not all models need a client. Right? I think this will fail for everything which is not Dataset.